### PR TITLE
Use the partial semantics workspace for the navbars to avoid expensive computation

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Precursor to potentially moving these parts of NavBar out of process.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@genlu @jmarolf thanks!

File: src/EditorFeatures/Core/Implementation/NavigationBar/NavigationBarController.cs [hunk 1]
```diff
@@ -382,7 +382,9 @@ private void ProcessItemSelectionSynchronously(NavigationBarItem item, Cancellat
             }
             else
             {
-                var document = _subjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+                // When navigating, just use the partial semantics workspace.  Navigation doesn't need the fully bound
+                // compilations to be created, and it can save us a lot of costly time building skeleton assemblies.
```
Comment (genlu) [inline src/EditorFeatures/Core/Implementation/NavigationBar/NavigationBarController.cs:386]: >skeleton assemblies [](start = 96, length = 19)

What is skeleton assembly?

**Review by genlu — APPROVED** [review]
I don't fully understand how navbar works, but the change itself makes sense to me

**Comment by jmarolf** [comment]
:shipit: late but the change looks good

**Pull request merged**